### ISSUE TITLE
feat: Baseline FHE primitives (LWE, rLWE)

### ISSUE THREAD

**Issue by 0xJepsen** [pluto/ronkathon#117]
# We should prepare for an FHE boom and get some base FHE primitives down.

I watched this lecture a while back and thought it was pretty good.

[LWE](https://www.youtube.com/watch?v=Lo-_ZBqGa7I&t=182s)

**Comment by mrdaybird** [comment]
I was thinking of implementing this paper (https://eprint.iacr.org/2013/340.pdf), this scheme is conceptually simple to understand. I think it will be a good starting point.

**Comment by Autoparallel** [comment]
@mrdaybird I'd be happy to work with you on these too!

**Comment by mrdaybird** [comment]
@Autoparallel Awesome! Let's have a chat over this over the weekend.

**Comment by Autoparallel** [comment]
Sure thing! Feel free to ping me wherever you want. @Autoparallel on Telegram

**Comment by Autoparallel** [comment]
This resource seems pretty decent for LWE: https://65610.csail.mit.edu/2024/lec/l07-pke.pdf

**Comment by mrdaybird** [comment]
@Autoparallel missed your messages, apparently all my github message got moved to spam :confused: 

> Want to join me?

definitely, what would you like me to do? 

> This resource seems pretty decent for LWE: https://65610.csail.mit.edu/2024/lec/l07-pke.pdf

I will through this resource in the meanwhile.

**Comment by mrdaybird** [comment]
Moving from a single bit to small messages would be interesting.
Also, it will be very interesting to think about how we will move from LWE encryption to a homomorphic encryption(maybe fully) scheme.

I will look into the latter.

**Comment by Autoparallel** [comment]
Yeah, I think the PR needs to change to allow encrypting larger payloads. I can look into this too.
